sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java

```diff
@@ -103,105 +82,58 @@ public void registerUdaf(String functionName, Combine.CombineFn combineFn) {
     schema.plus().add(functionName, new UdafImpl(combineFn));
   }
 
-  /**
-   * Registers {@link PCollection}s in {@link PCollectionTuple} as a tables.
-   *
-   * <p>Assumes that {@link PCollection} elements are {@link Row}s.
-   *
-   * <p>{@link TupleTag#getId()}s are used as table names.
-   */
-  public void registerPCollectionTuple(PCollectionTuple pCollectionTuple) {
-    pCollectionTuple
-        .getAll()
-        .forEach((tag, pCollection) ->
-                registerPCollection(tag.getId(), (PCollection<Row>) pCollection));
-  }
-
-  /**
-   * Registers {@link PCollection} of {@link Row}s as a table.
-   *
-   * <p>Assumes that {@link PCollection#getCoder()} returns an instance of {@link RowCoder}.
-   */
-  public void registerPCollection(String name, PCollection<Row> pCollection) {
-    registerTable(name, pCollection, ((RowCoder) pCollection.getCoder()).getSchema());
-  }
-
-  /**
-   * Registers {@link PCollection} as a table.
-   */
-  public void registerTable(String tableName, PCollection<Row> pCollection, Schema schema) {
-    registerTable(tableName, new BeamPCollectionTable(pCollection, schema));
-  }
-
-  /**
-   * Registers a {@link BaseBeamTable} which can be used for all subsequent queries.
-   */
-  public void registerTable(String tableName, BeamSqlTable table) {
-    schema.add(tableName, new BeamCalciteTable(table));
+  public BeamQueryPlanner getPlanner() {
+    return planner;
   }
 
-  public void deregisterTable(String targetTableName) {
-    schema.removeTable(targetTableName);
+  public CalcitePrepare.Context getContext() {
+    return new ContextImpl();
   }
 
-  private static class BeamCalciteTable extends AbstractQueryableTable
-      implements ModifiableTable, TranslatableTable {
-    private BeamSqlTable beamTable;
-
-    public BeamCalciteTable(BeamSqlTable beamTable) {
-      super(Object[].class);
-      this.beamTable = beamTable;
+  private class ContextImpl implements CalcitePrepare.Context {
+    @Override
+    public JavaTypeFactory getTypeFactory() {
+      throw new UnsupportedOperationException();
```

Comment:
Would it be wrong to return `BeamQueryPlanner.TYPE_FACTORY`?

Reply:
Nope, changed.